app/src/main/java/com/epfl/beatlink/repository/profile/FriendRequestRepositoryFirestore.kt

```diff
@@ -0,0 +1,173 @@
+package com.epfl.beatlink.repository.profile
+
+import android.util.Log
+import com.epfl.beatlink.model.profile.FriendRequestRepository
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FieldValue
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.tasks.await
+
+open class FriendRequestRepositoryFirestore(
```

Comment:
I think the implementation could benefit from introducing a data class, such as `FriendRequest`. This would improve readability and maintainability while also providing a clearer representation of the Firestore document structure.

For example:

```kotlin
data class FriendRequest(
    val ownRequests: Map<String, Boolean> = emptyMap(),
    val friendRequests: Map<String, Boolean> = emptyMap(),
    val allFriends: Map<String, FriendDetails> = emptyMap()
)

Reply:
Firestore naturally supports storing and retrieving maps directly, so creating I didn't think that a data class was necessary, but if we think about a clear structure for our data, it might have been better to create a data class. That said, adopting a data class at this stage would require significant changes to the existing functions. For now, the current implementation should work well without issues. If it becomes overly complex in the future, transitioning to a data class can be reconsidered.